src/target/riscv/riscv.h

```diff
@@ -152,6 +152,17 @@ struct riscv_info {
 	/* record the tinfo of each trigger */
 	unsigned int trigger_tinfo[RISCV_MAX_TRIGGERS];
 
+	/* record the tinfo_version of each trigger */
+	int64_t trigger_tinfo_version[RISCV_MAX_TRIGGERS];
```

Comment:
My interpretation of `tinfo.version`  has always been that it is a global constant, not a per-trigger constant. 

However, I am afraid that the debug spec is not clear about that, either. On page 69, the spec says:
- `The reset values listed here [in tinfo] apply to every underlying trigger.`
- `[Field version] contains the version of the Sdtrig extension implemented.`

These seem to contradict each other.

@en-sc What is your opinion/interpretation of this part of the spec, please?

Reply:
addressed

Reply:
@en-sc, thank you for your input. I have just submitted a proposal for the debug spec to clarify the description of tinfo.version: https://github.com/riscv/riscv-debug-spec/pull/1081. Feel free to comment on it.

@sunnyzhu-learning, please, update your code to read `tinfo.version` just once.

The tinfo in struct riscv_info should be just a scalar, not an array:

```
	/* Version of the implemented Sdtrig extension */
	int tinfo_version;
```

Please, also print the obtained tinfo.version value to the log (`LOG_TARGET_DEBUG`) so that it becomes easier to troubleshoot issues.

Reply:
addressed

Reply:
Please, the proper place for getting tinfo_version is not in `add_trigger()` but in `riscv_enumerate_triggers()`.

Reply:
addressed, getting tinfo_version move to `riscv_enumerate_triggers()`